src/amo/containers/CategoryPage.js

```diff
@@ -3,13 +3,13 @@ import React from 'react';
 import { connect } from 'react-redux';
 import { compose } from 'redux';
 
-import SearchPage from 'amo/components/SearchPage';
+import { SearchBase } from 'amo/components/Search';
 import { loadByCategoryIfNeeded, parsePage } from 'core/searchUtils';
 import { apiAddonType, safeAsyncConnect } from 'core/utils';
 
 
 export function CategoryPageBase(props) {
-  return <SearchPage enableSearchSort={false} {...props} />;
+  return <SearchBase enableSearchSort={false} {...props} />;
```

Comment:
I don't quite follow: that's already how we switched from Category and it was pretty straightforward; once we switch to saga it would be a matter of checking in `SearchBase` for the search results given the filter set. If that component always is set up to dispatch when it doesn't have the right search data things will be good as pages like `amo/components/Category` also use the `SearchBase` component to render out their data.

To be the separate component used to load the data is the one we export using compose. I don't think it should go in a separate file though.